modules/core/src/smithy4s/http/RawErrorResponse.scala

```diff
@@ -84,6 +84,18 @@ object RawErrorResponse {
   ): RawErrorResponse =
     new RawErrorResponse(code, headers, body, failedDecodeAttempt)
 
+  def unapply(response: RawErrorResponse): Option[
```

Comment:
**question:** do we want to keep the ability to use the `RawErrorResponse` as an extractor even though it might not contain all fields? 

Imo better just to make it private as suggested in the guide.

Reply:
Agreed

Reply:
The reason I made it non private because I got this error
```private method unapply in object RawErrorResponse is never used
[error] [core]   private def unapply(response: RawErrorResponse): Option[
[error] [core]               ^
[error] [core] No warnings can be incurred under -Werror.
```

Reply:
In that case, you can silence the warning: https://www.scala-lang.org/2021/01/12/configuring-and-suppressing-warnings.html#configuring-warnings